src/components/network/line-layer.js

```diff
@@ -195,336 +206,808 @@ class LineLayer extends CompositeLayer {
+                        getColor: (line) =>
+                            getLineColor(
+                                line,
+                                nominalVoltageColor,
+                                props,
+                                state.linesConnection.get(line.id)
+                            ),
+                        getWidth: 2,
+                        getProximityFactor: (line) =>
+                            isStart
+                                ? line.geo.proximityFactorStart
+                                : line.geo.proximityFactorEnd,
+                        getLineParallelIndex: (line) =>
+                            isStart
+                                ? line.geo.parallelIndex
+                                : -line.geo.parallelIndex,
+                        getLineAngle: (line) =>
+                            isStart
+                                ? line.geo.angleStart
+                                : line.geo.angleEnd + Math.PI,
+                        getDistanceBetweenLines: props.distanceBetweenLines,
+                        getMaxParallelOffset: props.maxParallelOffset,
+                        getMinParallelOffset: props.minParallelOffset,
+                        getSubstationRadius: props.substationRadius,
+                        getSubstationMaxPixel: props.substationMaxPixel,
+                        getMinSubstationRadiusPixel:
+                            props.minSubstationRadiusPixel,
+                        visible: props.filteredNominalVoltages.includes(
+                            nominalVoltage
+                        ),
+                        updateTriggers: {
+                            getLineParallelIndex: [
+                                state[PATH_UPDATED + nominalVoltageIndex],
+                            ],
+                            getSourcePosition: [
+                                state[PATH_UPDATED + nominalVoltageIndex],
+                            ],
+                            getTargetPosition: [
+                                state[PATH_UPDATED + nominalVoltageIndex],
+                            ],
+                            getLineAngle: [
+                                state[PATH_UPDATED + nominalVoltageIndex],
+                            ],
+                            getProximityFactor: [
+                                state[PATH_UPDATED + nominalVoltageIndex],
+                                state[PROXIMITY_UPDATED],
+                            ],
+                            getColor: [
+                                props.disconnectedLineColor,
+                                props.lineFlowColorMode,
+                                props.lineFlowAlertThreshold,
+                                props.updatedLines,
+                            ],
+                        },
+                    })
+                )
+        );
+        this.setState({ layers: layers }, () => this.scheduleNext(next));
```

Comment:
when is the lambda executed ? What is the value of next ?

Reply:
should not be there.